mobile/src/main/java/org/fedorahosted/freeotp/main/Adapter.java

```diff
@@ -285,7 +285,18 @@ public Code getCode(int position)
 
         try {
             Log.i(LOGTAG, String.format("getCode: deserialize token"));
-            Token token = Token.deserialize(mSharedPreferences.getString(uuid, null));
+            String stored = mSharedPreferences.getString(uuid, null);
+            Token token = null;
+            if (stored != null) {
+                token = Token.deserialize(stored);
+            } else {
+                Log.e(LOGTAG, "Stored is null");
+                return new Code("ERROR", 15);
+            }
+            if (token == null){
+                Log.e(LOGTAG, "Token is null");
+                return new Code("ERROR", 15);
+            }
             Key key = mKeyStore.getKey(uuid, null);
             code = token.getCode(key);
             mSharedPreferences.edit().putString(uuid, token.serialize()).apply();
```

Comment:
Note for myself, the token is updated in shared preferences as for HOTP counter value may have changed.